packages/blade/src/components/Modal/_decisions/decisions.md

```diff
@@ -0,0 +1,149 @@
+
+### `ModalFooter` API
+> `ModalFooter` API will be similar to [`BottomSheetFooter` API](https://github.com/razorpay/blade/blob/master/packages/blade/src/components/BottomSheet/_decisions/decisions.md#bottomsheetfooter-api).
+
+| Prop       | Type              | Default     | Description                 | Required |
+| ---------- | ----------------- | ----------- | --------------------------- | -------- |
+| `children` | `React.ReactNode` | `undefined` | Contents of the ModalFooter | ✅       |
+
+## Behaviors
+### Modal Stacking
+- We're at consensus that we do need to support modal stacking.
+- Design is not yet aligned on how the modal should look like when stacked and what the motion should be.
+- As an interim solution we can implement a basic stacking behavior with the same exit/entry animations as the current modal and look into enhancing this motion later.
+
+### Modal Sizes & Responsiveness
+- We will offer Modal in 3 sizes - small, medium and large.
+- Each size corresponds to a max-width for the Modal
+- As of today these are the max-widths we're using for each size:
+  - small: `400px`
+  - medium: `760px`
+  - large: `1024px`
+- All the Modals will have a min-width of `320px` regardless of the size.
+- This would mean that the Modal will be responsive and will scale down to `320px` on smaller screens but will not scale up beyond the max-width as per its size mentioned above.
+
+### Modal Height
+- Each Modal will have a max-height of `80vh` (80% of viewport height) and will be scrollable if the content exceeds this height.
+
+
+### Modal Usage on mWeb & native apps
+- We will not be using the Modal component on mWeb or native apps.
+- We will only use the BottomSheet component on mWeb and native apps.
+- Our Modal will be responsive and will not break up till 320px on mWeb but will not be usable on native apps.
+- We will throw a warning when a Modal is opened on smaller screens.
+- Designers would need to be mindful of this while designing for mWeb and native apps.
+- On Figma, we have added guardrails to ensure that when screenSize is selected Mobile on Figma, Modal component will turn into a BottomSheet component.
+- We are not automatically changing our Modal to BottomSheet on code to avoid bundle size overhead as well as to avoid any flow to break unless mindfully implemented with BottomSheet on code.
+
+### Modal for React Native
+- Since we want to restrict the usage of Modals on native apps, we will not be implementing the Modal component for React Native.
+- We will implement a dummy export which will only warn the consumer that this component is not supported on native apps.
+- We will also add this to our documentation.
+
+## Accessibility
+- Trap keyboard focus within the modal.
+- Close the modal when the user hits the `esc` key.
+- Add `aria-modal='true'`
+- Add `aria-role='dialog'`
+
+## Open questions
+- Should we have an upper limit on Modal stacking the same way we have an upper limit of 3 stacks on BottomSheet?
```

Comment:
We do not have upper limit of bottomsheet stack of 3. We just render sheets on top of them as is. 

This upper limit was only discussed when Abhishek did the exploration of stacking where the stacking animation & UX worked like [Vercel's toast](https://vercel.com/design/toast)

Reply:
Got it